scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/Rewrite.scala

```diff
@@ -85,7 +86,12 @@ object Rewrite {
   }
 
   def apply(input: Input, style: ScalafmtConfig): Input = {
-    val rewrites = style.rewrite.rules
+    val trailingCommaRewrite =
+      if (!style.runner.dialect.allowTrailingCommas ||
+        style.trailingCommas == TrailingCommas.preserve) Seq.empty
```

Comment:
If the `RewriteTrailingCommas` does only remove trailing commas, can we populate `RewriteTralingComma` only when `style.trailingCommas === TrailingCommas.never` ?

Reply:
unfortunately, we can't. the whole point is to remove idempotency issues, and that requires feeding the same exact code (ignoring whitespace, that is) into the formatter in successive iterations.

so, we have two options:
* in rewrite, remove unless `preserve`; then add in format if `always` and has newline, or
* in rewrite, remove for `never`, add for `always`; then remove in format if `always` and no newline.

the first option (the one i implemented) is simpler code but could potentially lead to `maxColumn+1` lines.

the second is a bit more code, probably less confusing in the rewrite step (don't need to explain why we are removing in case of `always`) and will not overflow lines.

@tanishiking can you indicate your preference, please?

Reply:
ah, one complication with the second option for `always` case which i just discovered.

since scala doesn't allow one-line `a(b, c,)` (unlike python), we must append the comma *and the newline* (if missing), which might mess with the config style logic.

oh well. i'd stay with option 1, then.

Reply:
next patch explains why we only remove, but the name (for now) is kept the same. i do not feel too strongly, let me know if after my explanation you'd still prefer to call it `RemoveTrailingCommas`. it will be invoked for `always` as well, though.

Reply:
Ah, I got it! As long as there's some explanation about this, I think we don't need to rename it :)
Can we add some comment on `val appendComma = ...` side that we have to drop trailing comma beforehand, to avoid idempotency case?

Others are LGTM:+1:

Reply:
will do shortly, thank you

Reply:
@tanishiking done. explanations added, regarding idempotence and appending commas in formatter. PTAL.